src/WebJobs.Script/Config/ScriptJobHostOptionsSetup.cs

```diff
@@ -59,6 +60,12 @@ public void Configure(ScriptJobHostOptions options)
             // FunctionTimeout
             ConfigureFunctionTimeout(options);
 
+            if (_environment.IsLogicApp())
+            {
+                // Logic Apps requires no timeout
+                options.MetadataProviderTimeout = Timeout.InfiniteTimeSpan;
```

Comment:
I get that you're defaulting this to Infinite to effectively remove this completely for LAs. However, this timeout was added presumably for a reason - to catch truly runaway cases. So I'm wondering if we shouldn't just configure a longer but still reasonable timeout here? E.g. anything on the order of minutes would likely be considered a broken LA from a customer point of view, and LA needs to optimize/fix their extension.;

Reply:
We had discussed a longer timeout period with the LA team, but they requested no timeout until they figure this out. An action item post this fix is to resolve that:

From Laveesh: `Long term, functions team does want to enforce a timeout. To ensure we honor the timeout properly, functions and logic apps PG teams will engage to figure out what the proper fixes should be on both our sides and work on them. This will be done independent of our response to this incident.`

Reply:
I've updated the PR to so that despite defaulting to infinite for logic apps, the configured value will be respected and can be overridden for logic apps as well.

After design conversations with LA later, we can get still change the LA default to a reasonable value instead of disabling it entirely (via infinite).